Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: test: CET/EET are deprecated, use Europe/Berlin and Kyiv
Origin: upstream, commit:aa077884c13769ae3bd6aa98978b4ac9e64b5365
diff --git a/src/test/test-calendarspec.c b/src/test/test-calendarspec.c
index 7cda114..54c39d6 100644
--- a/src/test/test-calendarspec.c
+++ b/src/test/test-calendarspec.c
@@ -185,18 +185,18 @@ TEST(calendar_spec_one) {
 
 TEST(calendar_spec_next) {
         test_next("2016-03-27 03:17:00", "", 12345, 1459048620000000);
-        test_next("2016-03-27 03:17:00", "CET", 12345, 1459041420000000);
-        test_next("2016-03-27 03:17:00", "EET", 12345, -1);
+        test_next("2016-03-27 03:17:00", "Europe/Berlin", 12345, 1459041420000000);
+        test_next("2016-03-27 03:17:00", "Europe/Kyiv", 12345, -1);
         test_next("2016-03-27 03:17:00 UTC", NULL, 12345, 1459048620000000);
         test_next("2016-03-27 03:17:00 UTC", "", 12345, 1459048620000000);
-        test_next("2016-03-27 03:17:00 UTC", "CET", 12345, 1459048620000000);
-        test_next("2016-03-27 03:17:00 UTC", "EET", 12345, 1459048620000000);
-        test_next("2016-03-27 03:17:00.420000001 UTC", "EET", 12345, 1459048620420000);
-        test_next("2016-03-27 03:17:00.4200005 UTC", "EET", 12345, 1459048620420001);
-        test_next("2015-11-13 09:11:23.42", "EET", 12345, 1447398683420000);
-        test_next("2015-11-13 09:11:23.42/1.77", "EET", 1447398683420000, 1447398685190000);
-        test_next("2015-11-13 09:11:23.42/1.77", "EET", 1447398683419999, 1447398683420000);
-        test_next("Sun 16:00:00", "CET", 1456041600123456, 1456066800000000);
+        test_next("2016-03-27 03:17:00 UTC", "Europe/Berlin", 12345, 1459048620000000);
+        test_next("2016-03-27 03:17:00 UTC", "Europe/Kyiv", 12345, 1459048620000000);
+        test_next("2016-03-27 03:17:00.420000001 UTC", "Europe/Kyiv", 12345, 1459048620420000);
+        test_next("2016-03-27 03:17:00.4200005 UTC", "Europe/Kyiv", 12345, 1459048620420001);
+        test_next("2015-11-13 09:11:23.42", "Europe/Kyiv", 12345, 1447398683420000);
+        test_next("2015-11-13 09:11:23.42/1.77", "Europe/Kyiv", 1447398683420000, 1447398685190000);
+        test_next("2015-11-13 09:11:23.42/1.77", "Europe/Kyiv", 1447398683419999, 1447398683420000);
+        test_next("Sun 16:00:00", "Europe/Berlin", 1456041600123456, 1456066800000000);
         test_next("*-04-31", "", 12345, -1);
         test_next("2016-02~01 UTC", "", 12345, 1456704000000000);
         test_next("Mon 2017-05~01..07 UTC", "", 12345, 1496016000000000);
@@ -215,7 +215,7 @@ TEST(calendar_spec_next) {
         test_next("2017-04-02 03:30:00 Pacific/Auckland", "", 12345, 1491060600000000);
         /* Confirm that timezones in the Spec work regardless of current timezone */
         test_next("2017-09-09 20:42:00 Pacific/Auckland", "", 12345, 1504946520000000);
-        test_next("2017-09-09 20:42:00 Pacific/Auckland", "EET", 12345, 1504946520000000);
+        test_next("2017-09-09 20:42:00 Pacific/Auckland", "Europe/Kyiv", 12345, 1504946520000000);
         /* Check that we don't start looping if mktime() moves us backwards */
         test_next("Sun *-*-* 01:00:00 Europe/Dublin", "", 1616412478000000, 1617494400000000);
         test_next("Sun *-*-* 01:00:00 Europe/Dublin", "IST", 1616412478000000, 1617494400000000);
diff --git a/test/test-functions b/test/test-functions
index 000d1fe..52167c7 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -2254,8 +2254,6 @@ install_zoneinfo() {
     inst_any /usr/share/zoneinfo/Europe/Kyiv
     inst_any /usr/share/zoneinfo/Pacific/Auckland
     inst_any /usr/share/zoneinfo/Pacific/Honolulu
-    inst_any /usr/share/zoneinfo/CET
-    inst_any /usr/share/zoneinfo/EET
     inst_any /usr/share/zoneinfo/UTC
 }
 
diff --git a/test/units/TEST-74-AUX-UTILS.firstboot.sh b/test/units/TEST-74-AUX-UTILS.firstboot.sh
index d9e5f59..79a9a77 100755
--- a/test/units/TEST-74-AUX-UTILS.firstboot.sh
+++ b/test/units/TEST-74-AUX-UTILS.firstboot.sh
@@ -132,7 +132,7 @@ systemd-firstboot --root="$ROOT" \
                   --locale=locale-overwrite \
                   --locale-messages=messages-overwrite \
                   --keymap=keymap-overwrite \
-                  --timezone=CET \
+                  --timezone=Europe/Berlin \
                   --hostname=hostname-overwrite \
                   --machine-id=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb \
                   --root-password-hashed="$ROOT_HASHED_PASSWORD2" \
@@ -153,7 +153,7 @@ systemd-firstboot --root="$ROOT" --force \
                   --locale=locale-overwrite \
                   --locale-messages=messages-overwrite \
                   --keymap=keymap-overwrite \
-                  --timezone=CET \
+                  --timezone=Europe/Berlin \
                   --hostname=hostname-overwrite \
                   --machine-id=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb \
                   --root-password-hashed="$ROOT_HASHED_PASSWORD2" \
@@ -162,7 +162,7 @@ systemd-firstboot --root="$ROOT" --force \
 grep -q "LANG=locale-overwrite" "$ROOT$LOCALE_PATH"
 grep -q "LC_MESSAGES=messages-overwrite" "$ROOT$LOCALE_PATH"
 grep -q "KEYMAP=keymap-overwrite" "$ROOT/etc/vconsole.conf"
-readlink "$ROOT/etc/localtime" | grep -q "/CET$"
+readlink "$ROOT/etc/localtime" | grep -q "/Europe/Berlin$"
 grep -q "hostname-overwrite" "$ROOT/etc/hostname"
 grep -q "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" "$ROOT/etc/machine-id"
 grep -q "^root:x:0:0:.*:/bin/barshell$" "$ROOT/etc/passwd"
